packages/shared/src/node-apis/redis.ts

```diff
@@ -127,12 +152,36 @@ export default class RedisClient {
     return value ? broadcastParsers[chain].parse(JSON.parse(value)) : null;
   }
 
-  async getDeposits(chain: Chain, asset: Asset, address: string) {
+  async getDeposits(chain: Chain, asset: Asset, address: string): Promise<PendingDeposit[]> {
     const parsedAddress = chain === 'Polkadot' ? ss58ToHex(address) : address;
     const key = `deposit:${chain}:${parsedAddress}`;
     const deposits = await this.client.lrange(key, 0, -1);
     return deposits
-      .map((deposit) => depositSchema.parse(deposit))
+      .map((deposit) => {
+        const parsedDeposit = depositSchema.parse(deposit);
+        const baseDeposit = {
+          amount: parsedDeposit.amount,
+          asset: parsedDeposit.asset,
+          deposit_chain_block_height: parsedDeposit.deposit_chain_block_height,
+        };
+        const { deposit_details, deposit_chain_block_height } = parsedDeposit;
+
+        if (!deposit_details) return { ...baseDeposit, tx_refs: [] };
+
+        switch (chain) {
```

Comment:
could we maybe reuse the `getDepositTxRef` method here?

Reply:
I've been trying to reuse that function but it didn't really work because the shape of the ingress egress tracker does not match the `depositDetails` types (`bitcoinSchema160`, `bitcoinSchema170`...)